File: 8kyu_repeat_str.py
def repeat_str(repeat, string):
    return "".join(string for n in range(0,repeat))

def repeat_str(repeat, string):
    return repeat * string
# ----------------------------------------------------------------------------
# !只能int*str 反過來會報錯

# import timeit
# def repeat_str_1():
#     repeat, string = 10, "aaa_"
#     return "".join(string for n in range(0,repeat))

# def repeat_str_2():
#     repeat, string = 10, "aaa_"
#     return repeat * string


# time_ori = timeit.timeit(repeat_str_1, number=100000)
# time_opt = timeit.timeit(repeat_str_2, number=100000)

# print(time_ori, time_opt)
# 0.09346470499440329 0.010097600999870338

File: 8kyu_repeat_str.java
class Solution {
    public static String repeatStr(final int repeat, final String string) {
        // StringBuilder arr = new StringBuilder();
        String[] arr = new String[repeat];
        for (int i = 0; i < repeat; i++) {
            arr[i] = string;
        }
        return String.join("", arr);
    }
}
// class Solution {
//     public static String repeatStr(final int repeat, final String string) {
//         StringBuilder arr = new StringBuilder();
//         for (int i = 0; i < repeat; i++) {
//             arr.append(string);
//         }
//         return arr.toString();
//     }
// }

// class Solution {
// static String repeatStr(int repeat, String string) {
// return string.repeat(repeat);
// }
// }

// Ori ver.: 0.029546 ms
// Ori ver.: 0.03547 ms
// Opt ver.: 0.007087 ms

// !最快還是 .repeat()